### PR TITLE
Connecting competition block to gladiator & some other fixes

### PR DESCRIPTION
**Pull request by deadlybutter**
- Connected the competition actions to gladiator
- Fixed ActionTypes props for when a competition is present (We handle Step proptypes elsewhere now)
- Updated competition block logic to account for the various states
- Fixed a bug with the notification component throwing an error when trying to hash a null key

### CODE REVIEW

File: resources/assets/components/ActionPage/index.js [hunk 1]
```diff
@@ -172,12 +172,7 @@ const ActionPage = (props) => {
 };
 
 ActionPage.propTypes = {
-  steps: PropTypes.arrayOf(PropTypes.shape({
-    title: PropTypes.string.isRequired,
-    content: PropTypes.string.isRequired,
-    background: PropTypes.string.isRequired,
-    photos: PropTypes.array,
-  })),
+  steps: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
```
Comment (DFurnes) [inline resources/assets/components/ActionPage/index.js]: This doesn't need to block merging, but I think this is a sign we're sending too arbitrary of a prop here.

File: resources/assets/actions/competition.js [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
-    // @TODO: Obviously, replace this with an api call.
-    setTimeout(() => {
-      const userId = getState().user.id;
-      dispatch({ type: JOINED_COMPETITION, campaignId, userId });
-    }, 2000);
+    (new Phoenix()).post('next/contests/users', {
+      legacyCampaignId: campaignId,
+      legacyCampaignRunId: campaignRunId,
+    }).then((response) => {
+      if (! response) throw new Error('competition signup failed');
```
Comment (weerd) [inline resources/assets/actions/competition.js]: Not needed for this merge but we could update to dispatch an action if contest signup failed and handle it somehow :)

File: resources/assets/actions/competition.js [hunk 3]
```diff
@@ -1,33 +1,47 @@
...
-// Action: check if the user joined the competition for the given campaign id
-export function checkForCompetition(campaignId) {
+// Action: check if the user joined the competition for the given campaign & run id
+export function checkForCompetition(campaignId, campaignRunId) {
   return (dispatch, getState) => {
     const userId = getState().user.id;
 
-    // @TODO: make api call
-    dispatch({ type: COMPETITION_FOUND, campaignId, userId });
+    (new Phoenix()).post('next/contests/users', {
```
Comment (weerd) [inline resources/assets/actions/competition.js]: This should be a `POST` request and also pass the `current_user=true` param.

File: resources/assets/actions/competition.js (hunk 2 above)
Comment (deadlybutter) [inline resources/assets/actions/competition.js]: @weerd Look below in the catch!

File: resources/assets/actions/competition.js (hunk 3 above)
Comment (deadlybutter) [inline resources/assets/actions/competition.js]: @weerd Thats why its returning the response for every new user i make... Thank you for spotting that

Comment (DFurnes) [inline resources/assets/actions/competition.js]: @weerd We removed `current_user=true` in #235!

File: resources/assets/actions/competition.js (hunk 2 above)
Comment (weerd) [inline resources/assets/actions/competition.js]: Oh gotcha, this is then caught below. Maybe then pass the message from this to the notification?

File: resources/assets/actions/competition.js (hunk 3 above)
Comment (weerd) [inline resources/assets/actions/competition.js]: oh yeah, duh! thanks @DFurnes I had a brain fart.

File: resources/assets/actions/competition.js (hunk 2 above)
Comment (deadlybutter) [inline resources/assets/actions/competition.js]: If we're adding more messaging to the site id wanna run it by Jen, the default error msg is a good catch all for now i think

File: resources/assets/actions/competition.js (hunk 3 above)
Comment (deadlybutter) [inline resources/assets/actions/competition.js]: So thats not it, back to the drawing board!

Comment (DFurnes) [inline resources/assets/actions/competition.js]: Sorry, @weerd's point about this needing to be a `GET` request is still true! Just commenting on the param.

Comment (deadlybutter) [inline resources/assets/actions/competition.js]: Agh - Bad copy paste, totally slipped by me there

**Comment by deadlybutter** [comment]
Just went through the entire flow again, seems like all is well! I think this is good for merge

**Comment by deadlybutter** [comment]
screen shot 2017-04-28 at 3 46 26 pm

double checking the waiting room end as well, seems good here

File: resources/assets/components/Notification/index.js [hunk 4]
```diff
@@ -28,7 +27,7 @@ const NotificationList = ({ notifications, removeNotification }) => (
   <div className="notification-list">
     {notifications.map(({ message, style }, index) => (
       <Notification
-        key={makeHash(message)}
+        key={index} // eslint-disable-line react/no-array-index-key
```
Comment (weerd) [inline resources/assets/components/Notification/index.js]: we'll eventually need to figure out a better solution for keys to avoid using indexes :\

Comment (deadlybutter) [inline resources/assets/components/Notification/index.js]: Agreed, but wanted to get around the error in the meantime!

**Pull request merged**